src/tikray/macropipe/registry.py

```
@@ -0,0 +1,22 @@
+import typing as t
+
+
+class Registry:
+    """Function registry for MacroPipe."""
+
+    r: t.Dict[str, t.Callable] = {}
```

Ruff (0.15.9)

[warning] Mutable default value for class attribute

(RUF012)